src/Models/ShallowWaterModels/solution_and_tracer_tendencies.jl

```diff
@@ -4,21 +4,15 @@ using Oceananigans.Coriolis
 using Oceananigans.Operators
 using Oceananigans.TurbulenceClosures: ∇_dot_qᶜ, ∂ⱼ_τ₁ⱼ, ∂ⱼ_τ₂ⱼ
 
-@inline half_g_h²(i, j, k, grid, h, g)  = @inbounds 1/2 * g * h[i, j, k]^2
+# bathymetry (hB) is assumed to be a negative value equal to - depth.
 @inline h_plus_hB(i, j, k, grid, h, hB) = @inbounds h[i, j, k] + hB[i, j, k]
 
-@inline x_pressure_gradient(i, j, k, grid, g, h, hB, formulation) = ∂xᶠᶜᶜ(i, j, k, grid, half_g_h², h, g)
-@inline y_pressure_gradient(i, j, k, grid, g, h, hB, formulation) = ∂yᶜᶠᶜ(i, j, k, grid, half_g_h², h, g)
+@inline x_pressure_gradient(i, j, k, grid, g, h, hB, ::ConservativeFormulation) = g * ℑxᶠᵃᵃ(i, j, k, grid, h) * ∂xᶠᶜᶜ(i, j, k, grid, h_plus_hB, h, hB)
+@inline y_pressure_gradient(i, j, k, grid, g, h, hB, ::ConservativeFormulation) = g * ℑyᵃᶠᵃ(i, j, k, grid, h) * ∂yᶜᶠᶜ(i, j, k, grid, h_plus_hB, h, hB)
```

Comment:
these lines deal with #3051, right?

Reply:
Yes

Reply:
Can we link this PR with that issue then. Edit the first comment pf the PR and write “Closes #3051”

Reply:
Thanks for the suggestion and done!